src/hooks/usePrimaryColorStyles.ts

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from 'react';
+import { PrimaryColorStyles } from '../types';
+import { rgbToHsl } from '../utils';
+
+type UsePrimaryColorStyles = (primaryColor?: string) => PrimaryColorStyles;
+
+const usePrimaryColorStyles: UsePrimaryColorStyles = (primaryColor) => {
+  const memoizedPrimaryColorStyles = useMemo(() => {
+    let primaryColorStyles = {
+      '--primary-color-h': '227',
+      '--primary-color-s': '70%',
+      '--primary-color-l': '46%',
+    };
+
+    if (primaryColor) {
+      // Extract r, g, b values from the string. (r), (g), (b)
+      const rgbRegex = /^\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})/;
+      const rgbMatch = primaryColor.trim().match(rgbRegex)?.slice(1);
+
+      if (rgbMatch && rgbMatch.length === 3) {
+        const [r, g, b] = rgbMatch;
+        const [h, s, l] = rgbToHsl(Number(r), Number(g), Number(b));
```

Comment:
I'm sorry I'm late to the party but why do we need hsl values? Can't we just use the rgb values? I think we can achieve the same thing for accent colors with the rgb() alpha channel. Am I missing something?